msteams-platform/graph-api/import-messages/update-import-messages.md

```diff
@@ -0,0 +1,373 @@
+    "lastModifiedDateTime": null,
+    "deleted": false,
+    "subject": null,
+    "summary": null,
+    "importance": "normal",
+    "locale": "en-us",
+    "policyViolation": null,
+    "from": {
+        "application": null,
+        "device": null,
+        "conversation": null,
+        "user": {
+            "id": "id-value",
+            "displayName": "Joh Doe",
+            "userIdentityType": "aadUser"
+        }
+    },
+    {
+      "body": {
+        "contentType": "html",
+        "content": "<div><div>\n<div><span><img height=\"250\" src=\"https://graph.microsoft.com/teams/teamId/channels/channelId/messages/id-value/hostedContents/hostedContentId/$value\" width=\"176.2295081967213\" style=\"vertical-align:bottom; width:176px; height:250px\"></span>\n\n</div>\n\n\n</div>\n</div>"
+    },
+    "attachments": [],
+    "mentions": [],
+    "reactions": []
+}
+```
+
+## Step Four: Complete migration mode
+
+Once the message migration process has completed, both the team and channel are taken out of migration mode using the  `completeMigration`  method. This step opens the team and channel resources for general use by team members. The action is bound to the `team` instance.
+
+#### Request (end team migration mode)
+
+```http
+POST https://graph.microsoft.com/beta/teams/teamId/completeMigration
+
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 NoContent
+```
+
+#### Request (end channel migration mode)
+
+```http
+POST https://graph.microsoft.com/beta/teams/teamId/channels/channelId/completeMigration
+
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 NoContent
+```
+
+#### Error response
+
+```http
+400 Bad Request
+```
+
+* Action called on a `team` or `channel` that is not in `migrationMode`.
+
+## Step Five: Add team members
+
+You can add a member to a team [using the Teams UI](https://support.microsoft.com/office/add-members-to-a-team-in-teams-aff2249d-b456-4bc3-81e7-52327b6b38e9) or Microsoft Graph [Add member](/graph/api/group-post-members?view=graph-rest-beta&tabs=http) API:
+
+#### Request (add member)
+
+```http
+POST https://graph.microsoft.com/beta/groups/{id}/members/$ref
+Content-type: application/json
+Content-length: 30
+
+{
+  "@odata.id": "https://graph.microsoft.com/beta/directoryObjects/{id}"
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+```
+
+## Tips and additional information
+
+<!-- markdownlint-disable MD001 -->
+<!-- markdownlint-disable MD026 -->
+
+* You can import messages from users who are not in Teams.
+
+* Once the `completeMigration` request is made, you cannot import further messages into the team.
+
+* Team members can only be added to the new team after the `completeMigration` request has returned a successful response.
+
+* Throttling: Messages import at 5 RPS per channel.
+
+* If you need to make a correction to the migration results, you need to delete the team and repeat the steps to create the team and channel and re-migrate the messages.
+
```

Comment:
The messages imported for users not present in tenant will not be searchable in Teams client or compliance portals for Public Preview